synapse/rest/admin/background_updates.py

```diff
@@ -104,4 +105,78 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                 "average_items_per_ms": update.average_items_per_ms(),
             }
 
-        return 200, {"enabled": enabled, "current_updates": current_updates}
+        return HTTPStatus.OK, {"enabled": enabled, "current_updates": current_updates}
+
+
+class BackgroundUpdateStartJobRestServlet(RestServlet):
+    """Allows to start specific background updates"""
+
+    PATTERNS = admin_patterns("/background_updates/start_job")
+    ERROR_DUPLICATE_JOB = "Job %s is already in queue of background updates."
+
+    def __init__(self, hs: "HomeServer"):
+        self._auth = hs.get_auth()
+        self._store = hs.get_datastore()
+
+    async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
+        requester = await self._auth.get_user_by_req(request)
+        await assert_user_is_admin(self._auth, requester.user)
+
+        body = parse_json_object_from_request(request)
+        assert_params_in_dict(body, ["job_name"])
+
+        job_name = body["job_name"]
+
+        if job_name == "populate_stats_process_rooms":
```

Comment:
Looks good! Can we rework this block a bit to reduce duplication. I think we can do:

```python
if job_name == "...":
    jobs = [...]
elif job_name == "...":
    jobs = [...]
else:
    raise SynError(...)

try:
    simple_insert_many(jobs...)
except IntegrityError:
    raise SynError(...)
```

We should be able to inline the error message too then.

Reply:
(And we'll end up using `simple_insert_many` even for the `populate_stats_process_rooms` but that's fine!)